tt_metal/tt_stl/any_range.hpp

```diff
@@ -0,0 +1,887 @@
+// SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include <algorithm>
+#include <iterator>
+#include <type_traits>
+#include <utility>
+
+namespace tt::stl {
+
+inline constexpr std::size_t any_iterator_alignment = alignof(void *);
+inline constexpr std::size_t any_range_alignment = alignof(void *);
+inline constexpr std::size_t default_any_iterator_capacity = 16;
+inline constexpr std::size_t default_any_range_capacity = 32;
```

Comment:
To have a bit more context, can you please provide the sizes of the common containers/iterators? I think it would be a bit better to define the default capacity as multiple of `sizeof(void*)`, like `2 * sizeof(void*)` / `4 * sizeof(void*)`.

Reply:
Yes, I understand that the values aren't dictated by the standard. But the code relies on these values being below the threshold in order to make the conversion, so it would be nice to know what are the sizes in practice.

Can you please write a comment here describing the consideration?